pkg/daemon/daemon.go

```diff
@@ -319,29 +353,28 @@ func (d *daemon) AddPeriodicUpdate() {
 		log.Info().Msgf("processing network networkID %s", networkID)
 		networkName, ibCniSpec, err := d.getIbSriovNetwork(networkID)
 		if err != nil {
-			if errcode.GetCode(err) == ErrNotIBSriovNetwork {
-				// Not an IB SR-IOV network
+			code := errcode.GetCode(err)
+			if code == ErrNotIBSriovNetwork || code == ErrGetNetAttachDef {
+				// Not an IB SR-IOV network or failed to get network attachment definition
 				addMap.UnSafeRemove(networkID)
+				log.Error().Msgf("droping network: %v", err)
+			} else {
+				log.Debug().Msgf("skipping network: %v", err)
```

Comment:
in this case we will retry this network in the next periodic update call right ?

Reply:
> in this case we will retry this network in the next periodic update call right ?

Right

> why would we want to do that ? in case of corrupt net-attach-def object ?

I assume for cases, when error wasn't persistent.